### ISSUE TITLE
Please provide propper examples! - How to secure segments, how to properly setup locations, how to use this outside of cloud providers etc etc ...

### ISSUE THREAD

**Issue by venomone** [kaltura/nginx-secure-token-module#96]
Hello,

I would be awesome to see some more examples around here, I personally want to stream segments using minio, well how to do that in a secure way? Not anyone want to use Akamai or Cloudflare nor AWS, we simply want it raw ;) ... Serving segments using VideoJS via a m3u8 playlist works for me if I just have one single location block configured like this:

       location /streaming {
           proxy_set_header Host $host;
           proxy_set_header X-Real-IP $remote_addr;
           proxy_pass http://minio0:9000/Movies;
       }

As soon as I setup something like this to secure my segments:

       location /streaming {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_pass http://minio0:9000/Movies;
        secure_token_encrypt_uri on;
        secure_token_encrypt_uri_key 0000000000000000000000000000000000000000000000000000000000000000;
     secure_token_encrypt_uri_iv 00000000000000000000000000000000;
     secure_token_types application/vnd.apple.mpegurl;

     secure_link $arg_md5;
     secure_link_md5 '$uri test';
     if ($secure_link = "") {
         return 404 "secure link error";
     }
     if ($secure_link = "0") {
         return 410;
     }

basically nothing works anymore. I also dont find anything at the docs that might describe what to do on application level? I'm using Django here.. Do I have to create any signed links using a secrets or what?
I'm kind of confused using the kaltura stack to be honest...
